x/batch/codec.proto

```diff
@@ -0,0 +1,48 @@
+syntax = "proto3";
+
+package batch;
+
+import "github.com/gogo/protobuf/gogoproto/gogo.proto";
+import "github.com/iov-one/weave/x/cash/codec.proto";
+import "github.com/iov-one/weave/x/escrow/codec.proto";
+import "github.com/iov-one/weave/x/namecoin/codec.proto";
+import "github.com/iov-one/weave/x/sigs/codec.proto";
+import "github.com/iov-one/weave/x/multisig/codec.proto";
+import "github.com/iov-one/weave/x/nft/codec.proto";
+import "github.com/iov-one/weave/x/nft/blockchain/codec.proto";
+import "github.com/iov-one/weave/x/nft/ticker/codec.proto";
+import "github.com/iov-one/weave/x/nft/username/codec.proto";
+import "github.com/iov-one/weave/x/nft/bootstrap_node/codec.proto";
+import "github.com/iov-one/weave/x/validators/codec.proto";
+
+// ExecuteBatchMsg encapsulates multiple messages to support batch transaction
+message ExecuteBatchMsg {
+    repeated Msg messages = 1 [(gogoproto.nullable) = false];
+    // Msg is a workaround as protobuf does not support repeated oneof
+    message Msg {
+        oneof msg {
```

Comment:
@alpe Since this is an extension I'm afraid we will have to deal with all the messages weave supports.
We are dealing with limitations of protobuf here mostly. 

It's super easy to fix this if we could forgo the protobuf codec for this particular case, but I'm afraid we have to be repetitive and verbose, look at how bnsd and bcpd codecs look for "inspiration".

Reply:
I think we will need to repeat somehow, but alex is right, this is application specific and should be in the `cmd/*/app` dir, not in batch. If in batch, then batch is tied to exactly one app and not re-usable.

Reply:
@ethanfrey makes sense, I always forget about the fact that one could extend weave outside it's codebase and add messages.
I will code that to some BatchMsg interface to allow defining these within the app.